clients/input_copy.go

```diff
@@ -95,6 +92,89 @@ func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVi
 	return
 }
 
+func isHLSInput(inputFile *url.URL) bool {
+	ext := strings.LastIndex(inputFile.Path, ".")
+	if ext == -1 {
+		return false
+	}
+	return inputFile.Path[ext:] == ".m3u8"
+}
+
+func getDestLoc(srcManifestUrl, dstTransferUrl *url.URL, srcSegmentUrl string) string {
+
+	parsedURL1, _ := url.Parse(srcManifestUrl.String())
+	parsedURL2, _ := url.Parse(srcSegmentUrl)
+
+	path1 := parsedURL1.Path
+	path2 := parsedURL2.Path
+
+	// Find the common prefix of the two paths
+	i := 0
+	for ; i < len(path1) && i < len(path2); i++ {
+		if path1[i] != path2[i] {
+			break
+		}
+	}
+
+	// Extract the relative path by removing the common prefix
+	relPath := path2[i:]
+	if strings.HasPrefix(relPath, "/") {
+		relPath = relPath[1:]
+	}
+
+	parsedURL3, _ := url.Parse(dstTransferUrl.String())
+	dir := path.Dir(parsedURL3.Path)
+	u := parsedURL3.User.Username()
+	pass, _ := parsedURL3.User.Password()
+	p := url.QueryEscape(pass)
+
+	parsedURL3.User = url.UserPassword(u, p)
+
+	newURL := parsedURL3.Scheme + "://" + u + ":" + p + "@" + parsedURL3.Host + dir + "/" + relPath
+
+	return newURL
+}
+
+func copyFile(requestID string, srcInputUrl, dstOutputUrl *url.URL) (size int64, err error) {
```

Comment:
Will this also work if the input file is the `s3+https://` URL. E.g. `s3+https://<ACCESS_KEY>:<SECRET_KEY>@gateway.storjshare.io/testinput/video.mp4`

Reply:
I'll test and confirm but it should b/c any mp4 input will not go through any of the new code paths being introduced here.

Reply:
Yes, this will work as long as the s3+https url has user creds.